test/common/crudtester.go

```diff
@@ -564,8 +564,10 @@ func (c *FederatedTypeCrudTester) waitForResource(client util.ResourceClient, qu
 
 			// Validate that the expected override was applied
 			if len(expectedOverrides) > 0 {
+				// TODO: use the json patch library to patch with the overrides and then compare it with cluster object.
```

Comment:
The CI failure suggests that you may need to implement this strategy to support testing of overriding a path that embeds an index.  The nested field copy won't work for that case.